internal/service/advancedclustertpf/validate_schema.go

```diff
@@ -0,0 +1,76 @@
+package advancedclustertpf
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var defaultMongoDBMajorVersion = "8.0"
+
+func PlanMustUseMongoDBVersion(version float64, mustBeLower bool) FailOnIncompatibleMongoDBVersion {
```

Comment:
nit: maybe using some predefined operators would make this more intuitive to use